bindings/python/tests/test_collection.py

```diff
@@ -159,6 +161,24 @@
 assert isinstance(coll["meta"], dlite.Metadata)
 
 
+# Test Collection.value()
+assert coll.value(s="dog", p="is-a") == "animal"
+assert coll.value(p="_is-a", o="Instance", any=1) == "inst1"
+assert coll.value(p="is-a", o="x", default="y", any=1) == "y"
+
+with raises(dlite.DLiteTypeError):
+    coll.value(s="dog", p="is-a", o="animal")
+
+with raises(dlite.DLiteTypeError):
+    coll.value()
+
+with raises(dlite.DLiteSearchError):
+    coll.value(s="dog", p="x")
+
+with raises(dlite.DLiteSearchError):
+    coll.value(p="_is-a", o="Instance")
```

Comment:
What happens if we set d? Do we get a nice error? I think that is something that people could think of doing since we can set s, p, o.

Reply:
Added two additional tests for it. Seems to work :-)

```python
assert coll.value(s="inst1", p="_has-uuid", d="xsd:anyURI") == (
    "de1eabc0-8f8e-4a64-9cd6-74ea4b2e3c54"
)
with raises(dlite.DLiteSearchError):
    coll.value(s="inst1", p="_has-uuid", d="xsd:int")
```